docs/source/reference/collectors.rst

```diff
@@ -59,6 +59,49 @@ Besides those compute parameters, users may choose to configure the following pa
 - exploration_type: the exploration strategy to be used with the policy.
 - reset_when_done: whether environments should be reset when reaching a done state.
 
+Collectors and batch size
+-------------------------
+
+Because each collector has its own way of organizing the environments that are
+run within, the data will come with different batch-size depending on how
+the specificities of the collector. The following table summarizes what is to
+be expected when collecting data:
+
+
++--------------------+---------------------+-----------------------------+------------------------------+
+|                    | SyncDataCollector   |MultiSyncDataCollector (n=B) |MultiaSyncDataCollector (n=B) |
++====================+=====================+=============+===============+==============================+
+|   `stack_result`   |          NA         |    `True`   |    `False`    |             NA               |
++--------------------+---------------------+-------------+---------------+------------------------------+
+|     Single env     |         [T]         |   `[B, T]`  |  `[B*(T//B)`  |              [T]             |
++--------------------+---------------------+-------------+---------------+------------------------------+
+| Batched env (n=P)  |       [P, T]        | `[B, P, T]` |  `[B * P, T]` |            [P, T]            |
```

Comment:
`[B * P, T]`, this is given that ` cat_results=0`

my personal preference, to avoid having `cat_results` be both int and str is to have 2 args
- `cat_results` true or false
- `collectors_dim` dimension where to cat or stack

Reply:
but then what is `collector_dim=0` and `cat_results=False`? That should not be allowed. So we need a complicated doc that lists the available configs
If we allow them we also need to test every single combination...

Reply:
Note that the goal is to get rid of this arg in v0.6 or v0.7 so adding 2 args is twice as bad as adding 1 given that we want to break things on the long run

Reply:
> collector_dim=0 and cat_results=False

isn't this what happens currently when `cat_results="stack"`?

Anyway yes how you prefer

Reply:
Yes sorry, I meant collector_dim=-1 or 1 or anything that isn't 0

you can't stack along any dim (and should not be able to do so), that's what I mean. Stack should be along dim 0

Reply:
i don't super see this as in the example table we are referring to, in the row for stacking in multisync i can see both

- B, P, T (collector_dim=0 or -3, cat_results=False) (currently the only available option)
- P, B, T (collector_dim=1 or -2, cat_results=False)
- P, T, B (collector_dim=2 or -1, cat_results=False)

Reply:
There is no `cat_results=False`, it can only be a string ("stack") or an int
cat_results="stack" => stack B collectors along 0 => [B, P, T]
cat_results=0 => cat B collectors along 0 => [B * P, T]
cat_results=-1 => cat B collectors along -1 => [B, P * T]

Reply:
I think I didn't get what you were trying to say though

Reply:
I know, that is the point of this thread, I am discussing why for me it makes sense to separate them.

Reply:
> P, B, T (collector_dim=1 or -2, cat_results=False)
> P, T, B (collector_dim=2 or -1, cat_results=False)

I don't think we should support these. If you want that you can transpose your resulting tensordict. Implementing it would require a lot of expensive tests for a feature we will deprecate in 2 releases.